third_party/xla_client/xrt_computation_client.h

```diff
@@ -114,18 +114,28 @@ class XrtComputationClient : public ComputationClient {
       return task_no == rhs.task_no && name == rhs.name;
     }
 
+    std::string ToString() const {
+      return name + ":" + std::to_string(task_no);
+    }
+
+    friend std::ostream& operator<<(std::ostream& stream,
+                                    const Worker& worker) {
+      stream << worker.ToString();
+      return stream;
+    }
+
     std::string name;
     int task_no;
   };
 
   struct Options {
     std::string default_device;
-    // Maps a PyTorch device ID (example, "GPU:0", "TPU:0") to the full
+    // Maps a PyTorch Worker (example, {"GPU", 0}, {"TPU", 0}) to the full
     // coordinates in TF device format
     // (ie, /job:tpu_worker/replica:0/task:0/device:TPU:0), of the worker
     // exposing that device. These devices are all the devices present within
     // the TPU mesh.
-    std::map<std::string, std::string> global_device_map;
+    std::map<Worker, std::string> global_device_map;
```

Comment:
This is not the correct fix.
The key should be the `Device` class defined at line 94, to which an `operator<()` should be added (as well as a `ToString()` and streaming `<<` eventually).
The key is a *Device* not a *Worker*.

Reply:
Ah thanks for pointing out. I now see `Worker` is more of a TF worker concept it seems. I've updated to use `Device`.